### ISSUE TITLE
XunitTestCollectionRunner - IAsyncLifetime's DisposeAsync and IDisposable's Dispose run in parallel

### ISSUE THREAD

**Issue by maxcherednik** [xunit/xunit#1749]
**Summary:**

When a test collection implements both IAsyncLifetime and IDisposable, corresponding Dispose methods are executed in parallel despite the documentation:
```C#
     /// <summary>
    /// Called when an object is no longer needed. Called just before <see cref="M:System.IDisposable.Dispose" />
    /// if the class also implements that.
    /// </summary>
    Task DisposeAsync();
```

**Source:**
https://github.com/xunit/xunit/blob/89d11eca58bed6754304a2b3897bf2a0f6dcd838/src/xunit.execution/Sdk/Frameworks/Runners/XunitTestCollectionRunner.cs#L56

**Workaround**
Just use one or another interface

**Comment by bradwilson** [comment]
This is a bug in implementation, rather than documentation. It is intended that these would be run sequentially, not in parallel.
